src/cmd/config.zig

```diff
@@ -67,53 +71,43 @@ const GossipConfig = struct {
         };
     }
 
-    pub fn getNetwork(self: GossipConfig) error{UnknownNetwork}!?Network {
+    pub fn getCluster(self: GossipConfig) error{UnknownCluster}!?Cluster {
         return if (self.network) |network_str|
-            std.meta.stringToEnum(Network, network_str) orelse
-                error.UnknownNetwork
+            std.meta.stringToEnum(Cluster, network_str) orelse
+                error.UnknownCluster
         else
             null;
     }
-};
 
-pub const Network = enum {
-    mainnet,
-    devnet,
-    testnet,
-    localnet,
-
-    const Self = @This();
-
-    pub fn entrypoints(self: Self) []const []const u8 {
-        return switch (self) {
-            .mainnet => &.{
-                "entrypoint.mainnet-beta.solana.com:8001",
-                "entrypoint2.mainnet-beta.solana.com:8001",
-                "entrypoint3.mainnet-beta.solana.com:8001",
-                "entrypoint4.mainnet-beta.solana.com:8001",
-                "entrypoint5.mainnet-beta.solana.com:8001",
-            },
-            .testnet => &.{
-                "entrypoint.testnet.solana.com:8001",
-                "entrypoint2.testnet.solana.com:8001",
-                "entrypoint3.testnet.solana.com:8001",
-            },
-            .devnet => &.{
-                "entrypoint.devnet.solana.com:8001",
-                "entrypoint2.devnet.solana.com:8001",
-                "entrypoint3.devnet.solana.com:8001",
-                "entrypoint4.devnet.solana.com:8001",
-                "entrypoint5.devnet.solana.com:8001",
-            },
-            .localnet => &.{
-                "127.0.0.1:1024", // agave test-validator default
-                "127.0.0.1:8001", // sig validator default
-            },
-        };
+    pub fn getEntrypointAddrs(
+        self: GossipConfig,
+        allocator: std.mem.Allocator,
+    ) ![]SocketAddr {
+        var entrypoint_set = std.AutoArrayHashMap(SocketAddr, void).init(allocator);
+        defer entrypoint_set.deinit();
+
+        // add cluster entrypoints
+        if (try self.getCluster()) |cluster| {
+            for (sig.gossip.getClusterEntrypoints(cluster)) |entrypoint| {
+                const socket_addr = try resolveSocketAddr(allocator, entrypoint);
```

Comment:
It's a little unexpected to find network io in this file, but I guess it's fine.